engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/OauthTransform.kt

```diff
@@ -0,0 +1,87 @@
+package graphql.nadel.enginekt.transform
+
+import graphql.GraphQLError
+import graphql.introspection.Introspection
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.NadelExecutionContext
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.transform.query.NadelQueryTransformer
+import graphql.nadel.enginekt.transform.query.QueryPath
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.nadel.enginekt.transform.result.json.JsonNodeExtractor
+import graphql.nadel.enginekt.util.queryPath
+import graphql.normalized.ExecutableNormalizedField
+import graphql.validation.ValidationError
+import graphql.validation.ValidationErrorType
+
+class OauthTransform : NadelTransform<GraphQLError> {
+
+    override suspend fun isApplicable(
+        executionContext: NadelExecutionContext,
+        executionBlueprint: NadelOverallExecutionBlueprint,
+        services: Map<String, Service>,
+        service: Service,
+        overallField: ExecutableNormalizedField
+    ): GraphQLError? {
+        if (overallField.objectTypeNames
+                .map { executionBlueprint.schema.getType(it) }
+                .any { it == null }
+        ) {
+            return null
+        }
+
+        if (overallField.getOneFieldDefinition(executionBlueprint.schema)
+                .getDirective("scopes") != null
```

Comment:
Change this to some random directive name for now, or implement it only in tests.

Reply:
moved the transform to the test folder and I am feeding the transform to the engine in the `TestFixture.kt`. Let me know what you think